enterprise/server/usage/usage.go

```diff
@@ -83,41 +128,235 @@ func (ut *tracker) Increment(ctx context.Context, uc *tables.UsageCounts) error
 		return nil
 	}
 
-	t := ut.clock.Now()
+	t := ut.currentCollectionPeriod()
 
 	pipe := ut.rdb.TxPipeline()
 	// Add the group ID to the set of groups with usage
 	groupsCollectionPeriodKey := groupsRedisKey(t)
 	pipe.SAdd(ctx, groupsCollectionPeriodKey, groupID)
 	pipe.Expire(ctx, groupsCollectionPeriodKey, redisKeyTTL)
 	// Increment the hash values
-	groupIDCollectionPeriodKey := countsRedisKey(groupID, t)
+	countsKey := countsRedisKey(groupID, t)
 	for countField, count := range counts {
-		pipe.HIncrBy(ctx, groupIDCollectionPeriodKey, countField, count)
+		pipe.HIncrBy(ctx, countsKey, countField, count)
 	}
-	pipe.Expire(ctx, groupIDCollectionPeriodKey, redisKeyTTL)
+	pipe.Expire(ctx, countsKey, redisKeyTTL)
 	if _, err := pipe.Exec(ctx); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func collectionPeriodStart(t time.Time) time.Time {
+// StartDBFlush starts a goroutine that periodically flushes usage data from
+// Redis to the DB.
+func (ut *tracker) StartDBFlush() {
+	go func() {
+		ctx := context.Background()
+		for {
+			select {
+			case <-time.After(flushInterval):
+				if err := ut.FlushToDB(ctx); err != nil {
+					log.Errorf("Failed to flush usage data to DB: %s", err)
+				}
+			case <-ut.stopFlush:
+				return
+			}
+		}
+	}()
+}
+
+// StopDBFlush cancels the goroutine started by StartDBFlush.
+func (ut *tracker) StopDBFlush() {
+	ut.stopFlush <- struct{}{}
+}
+
+// FlushToDB flushes usage metrics from any finalized collection periods to the
+// DB.
+//
+// Public for testing only; the server should call StartDBFlush to periodically
+// flush usage.
+func (ut *tracker) FlushToDB(ctx context.Context) error {
+	// redsync.ErrFailed can be safely ignored; it just means that another app
+	// is already trying to flush.
+	if err := ut.flushToDB(ctx); err != nil && err != redsync.ErrFailed {
+		return err
+	}
```

Comment:
nit: since you're doing something special here, it's probably clearer to write it out:
err := ut.flushToDB(ctx)
if err != nil && err != redsync.ErrFailed {
  return err
}
return nil